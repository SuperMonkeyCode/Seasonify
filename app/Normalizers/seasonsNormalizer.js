function seasonsNormalizer(seasons) {
  return seasons.reduce((obj, season) => {
    if (season.seasonNumber === 0) {
      return obj;
    }
    return {
      ...obj,
      [parseInt(season.seasonNumber, 10)]: {
        id: season.seasonNumber,
        year: season.yearEnd,
        episodes: season.episodeCount,
        allocine: season.code,
      },
    };
  }, {});
}

export default seasonsNormalizer;
