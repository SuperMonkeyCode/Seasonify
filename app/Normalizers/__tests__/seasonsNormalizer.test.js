import seasonsNormalizer from '../seasonsNormalizer';

const input = require('../../Fixtures/apiSeasonsDeadwood.json');

test('seasonsNormalizer output should match snapshot (included a boggus "0" season)', () => {
  expect(seasonsNormalizer([...input, { seasonNumber: 0 }])).toMatchSnapshot();
});
