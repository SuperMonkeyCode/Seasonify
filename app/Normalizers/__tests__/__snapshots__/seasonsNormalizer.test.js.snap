// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seasonsNormalizer output should match snapshot (included a boggus "0" season) 1`] = `
Object {
  "1": Object {
    "allocine": 565,
    "episodes": 12,
    "id": 1,
    "year": 2004,
  },
  "2": Object {
    "allocine": 711,
    "episodes": 12,
    "id": 2,
    "year": 2005,
  },
  "3": Object {
    "allocine": 2664,
    "episodes": 12,
    "id": 3,
    "year": 2006,
  },
}
`;
